numba/core/typing/builtins.py

```diff
@@ -954,7 +954,7 @@ def generic(self, args, kws):
         if kws:
             raise errors.NumbaAssertionError('kws not supported')
 
-        [arg] = args
+        arg = args[0]
```

Comment:
Can you add a comment here saying that support for `int(str)` is done via `@overload`